brave/src/main/java/brave/propagation/ExtraField.java

```diff
@@ -0,0 +1,409 @@
+ * the License.
+ */
+package brave.propagation;
+
+import brave.internal.Nullable;
+import brave.internal.PropagationFields;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import static brave.propagation.CorrelationFields.validateName;
+import static brave.propagation.ExtraFieldPropagation.currentTraceContext;
+import static java.util.Arrays.asList;
+
+/**
+ * Defines a request-scoped field, usually but not always analogous to an HTTP header. Fields will
+ * be no-op unless {@link ExtraFieldPropagation} is configured.
+ *
+ * <p>For example, if you have a need to know a specific request's country code, you can
+ * propagate it through the trace:
+ * <pre>{@code
+ * // Configure your extra field
+ * countryCode = ExtraField.create("country-code");
+ *
+ * // If configured and set, you can retrieve it later. All of the below result in the same tag:
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue());
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue(context));
+ * spanCustomizer.tag("country-code", ExtraField.getValue("country-code"));
+ *
+ * // You can also update the value similarly, so that the new value will propagate downstream.
+ * countryCode.updateValue("FO");
+ * countryCode.updateValue(context, "FO");
+ * ExtraField.updateValue("country-code", "FO");
+ * ExtraField.updateValue(context, "country-code", "FO");
+ * }</pre>
+ *
+ * <h3>Correlation</h3>
+ * If you want an extra field to also be available in correlation such as logging contexts, use
+ * {@link Builder#withCorrelation()}.
+ *
+ * <pre>{@code
+ * // configure the field, permitting it to be used in correlation contexts
+ * amznTraceId = ExtraField.newBuilder("x-amzn-trace-id").withCorrelation().build();
+ *
+ * // Allow logging patterns like %X{traceId} %X{x-amzn-trace-id}
+ * loggingContext = new Log4J2Context();
+ *
+ * decorator = CorrelationFieldScopeDecorator.newBuilder(loggingContext)
+ *                                           .addField(amznTraceId).build();
+ *
+ * tracingBuilder.propagationFactory(ExtraFieldPropagation.newFactoryBuilder(B3Propagation.FACTORY)
+ *                                                        .addField(amznTraceId)
+ *                                                        .build())
+ *               .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+ *                                                                  .addScopeDecorator(decorator)
+ *                                                                  .build())
+ * }</pre>
+ *
+ * <h3>Prefixed fields</h3>
+ * <p>You can also prefix fields, if they follow a common pattern. For example, the following will
+ * propagate the field "x-vcap-request-id" as-is, but send the fields "country-code" and "user-id"
+ * on the wire as "baggage-country-code" and "baggage-user-id" respectively.
```

Comment:
Realized that baggage + country code is fun too

Reply:
these are actual header names prefixes. ex sleuth adds the baggage- prefix. you'll find a lot where we intentionally use real examples so that things aren't so abstract. This isn't just test json in zipkin, rather all the projects.

Is there a prefix you've mapped that would be better I tried to find a header prefixed userinfo and couldn't..